vivarium/interface/interactive.py

```diff
@@ -0,0 +1,120 @@
+from math import ceil
+
+from vivarium.framework.configuration import build_simulation_configuration, build_model_specification
+from vivarium.framework.plugins import PluginManager
+from vivarium.framework.engine import SimulationContext
+
+from .utilities import run_from_ipython, log_progress
+
+
+class InteractiveContext(SimulationContext):
+
+    def __init__(self, configuration, components, plugin_manager=None):
+        super().__init__(configuration, components, plugin_manager)
+        self._initial_population = None
+
+    def setup(self):
+        super().setup()
+        self._start_time = self.clock.time
+
+    def initialize_simulants(self):
+        super().initialize_simulants()
+        self._initial_population = self.population.population
+
+    def reset(self):
+        # This is super crude, but should work for a great deal of components.
```

Comment:
Yeah, it's real hard to check for internal state in components. It could actually keep a record of the initial component configuration and recreate the component manager from scratch.

Reply:
I'm considering many of these features in essentially an alpha state of development.  This seems useful.  I don't want to jump into the more heavy weight solution of essentially rebuilding the simulation without some more thought.

I want some user testing to see how it breaks.  I think it's worth pushing against components maintaining state info outside the state table.  Can we categorize and generalize that behavior?